lib/config/configitem.cpp

```diff
@@ -478,7 +476,11 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 					return;
 				}
 
+				itemsCount++;
 				committed_items++;
```

Comment:
You wouldn't have to capture `itemsCount` in the lambda if instead of incrementing it here each time, you just do `itemsCount += committed_items` after `upq.Join()`. Then `itemsCount` doesn't even have to be atomic.

(Treat this comment as FYI, if you like the suggestion, feel free to include it, but I don't expect you to change this, the current code is fine as well)